### PR TITLE
Setting up associations

### PR DESCRIPTION
**Pull request by iankiema**
The following files have been added  and updated
-Book.rb,
-Classroom.rb,
-Rental.rb 
-Person.rb,
-Student.rb
-Tests.rb

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 require_relative 'person'
 
 class Student < Person
```
Comment (zeff96) [inline student.rb:3]: - [ ] Hello! When setting up a classroom for a student, ensure that the student is also added to the list of students in that classroom. We should implement a method that establishes the 'belongs-to' relationship, indicating that a student belongs to a specific classroom. 📚👩‍🏫

```suggestion
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (zeff96) [inline rental.rb]: - [ ] Hello, kindly make sure that Getters and Setters for @book and @person instance variables are present

```sh
attr_accessor :date, :book, :person
```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals << rental
```
Comment (zeff96) [inline book.rb]: - [ ] Hello, Kindly use `@rentals` instance variable

File: person.rb [hunk 4]
```diff
@@ -29,4 +30,8 @@ def of_age?
   def generate_id
     rand(1..1000)
   end
+
+  def add_rental(rental)
+    rentals << rental
```
Comment (zeff96) [inline person.rb]: - [ ] Hello, Kindly use `@rentals` instance variable

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @iankiema , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost


## Highlights
✔️ Linters passing
✔️ Correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @iankiema, 

## Highlights!
- Good code organization and associations ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**